packages/flutter_test/lib/src/binding.dart

```diff
@@ -119,6 +119,20 @@ mixin TestDefaultBinaryMessengerBinding on BindingBase, ServicesBinding {
 /// that actually needs to make a network call should provide its own
 /// `HttpClient` to the code making the call, so that it can appropriately mock
 /// or fake responses.
+///
+/// ### Coordinate spaces
+///
+/// [TestWidgetsFlutterBinding] might be run on devices of different screen
+/// sizes, while the testing widget is still told the same size to ensure
+/// consistent results. Consequently, code that deals with positions (such as
+/// pointer events or painting) must distinguish between two coordinate spaces:
+///
+///  * The _local coordinate space_ is the one used by the testing widget
+///    (typically an 800 by 600 window, but can be altered by [setSurfaceSize]).
+///  * The _global coordinate space_ is the one used by the device.
+///
+/// Positions can be transformed between coordinate spaces with [localToGlobal]
+/// and [globalToLocal].
```

Comment:
Nice explanation! 👍

Reply:
Maybe add a note that global and local coordinate space are usually identically when run in the headless flutter_tester.

Reply:
Really? I thought they are still 800x600 and 2400x1800 respectively, which still require transforming. Also I worry if that comment means people don't need to worry about transforming in normal cases - they always should!

Reply:
Ok, let's leave the comment as is.